packages/discussions/src/utils/channels/can-post-to-channel.ts

```diff
@@ -44,11 +52,148 @@ export function canPostToChannel(
   });
 }
 
+function isAuthorizedToPostByChannelAcl(
+  user: IDiscussionsUser,
+  channelAcl: IChannelAclPermission[]
+): boolean {
+  const permissions = mapPermissionsByCategory(channelAcl);
+
+  if (canAnyUserPost(permissions[AclCategory.ANONYMOUS_USER])) {
+    return true;
+  }
+
+  if (user.username === null) {
+    return false;
+  }
+
+  return (
+    canAnyAuthenticatedUserPost(permissions[AclCategory.AUTHENTICATED_USER]) ||
+    canThisUserPost(user, permissions[AclCategory.USER]) ||
+    isAllowedToPostByUsersGroups(user, permissions[AclCategory.GROUP]) ||
+    isAllowedToPostByUsersOrg(user, permissions[AclCategory.ORG])
+  );
```

Comment:
I'm finding this a little hard to grok.  Can we rephrase to `aclAllowsAnyAuthenticatedUserToPost`, `aclAllowsThisUserToPost`, `aclAllowsThisUserToPostByGroup`, `aclAllowsThisUserToPostByOrg`?